components/collector/src/source_collectors/dependency_track/source_up_to_dateness.py

```diff
@@ -28,16 +20,19 @@ async def _api_url(self) -> URL:
 
     async def _parse_source_response_date_time(self, response: Response) -> datetime:
         """Override to parse the timestamp from the response."""
-        projects = await response.json(content_type=None)
-        datetimes = [self._last_bom_import_datetime(project) for project in projects]
-        return self.minimum(datetimes)
+        if datetimes := [
+            self._last_bom_import_datetime(project) async for project in self._get_projects_from_response(response)
+        ]:
+            return self.minimum(datetimes)
+        error_message = "No projects found"
+        raise CollectorError(error_message)
```

Comment:
Why raise an error, don't we usually return no-data when filters don't match? Is it a solution to use something like `datetime.MIN`, or should this code not be reached when the filter doesn't match?

Reply:
"Give me the date time the items in this empty set were last updated" is non-sensical, hence the error. There's no "no-data" when the metric is a date time. Returning `datetime.MIN` just leads to confused users.